### PR TITLE
Add rgb view in aditof-demo

### PR DESCRIPTION
**Pull request by SeptimiuVana**

### CODE REVIEW

File: examples/aditof-demo/aditofdemoview.h [hunk 1]
```diff
@@ -91,6 +89,12 @@ class AdiTofDemoView {
     bool localModeChecked = false;
     bool ethModeChecked = false;
     int connectionCurrentValue = 4;
+
+    bool depthIrChecked = true;
+    bool depthOnlyChecked = false;
+    bool irOnlyChecked = false;
+    int frameTypeCurrentValue = 4;
+    // 4 = depthIr(default); 2 = depthOnly; 1 = irOnly
```
Comment (dNechita) [inline examples/aditof-demo/aditofdemoview.h:101]: can you put this comment on the same line as the line code?

**Comment by dNechita** [comment]
There is a lot of code that is enabled only if smart 3D camera is enabled. Wouldn't it be possible to analyze the frame type of the camera and if it contains RBG content then do all the call to be able to display the RGB image?

**Comment by SeptimiuVana** [comment]
> There is a lot of code that is enabled only if smart 3D camera is enabled. Wouldn't it be possible to analyze the frame type of the camera and if it contains RBG content then do all the call to be able to display the RGB image?

We could but it will add a few more conditions to be verified if the 3D camera is disabled.

File: examples/aditof-demo/aditofdemoview.h (hunk 1 above)
Comment (dNechita) [reply examples/aditof-demo/aditofdemoview.h:101]: yes, that might be a little better

**Comment by dNechita** [comment]
> > There is a lot of code that is enabled only if smart 3D camera is enabled. Wouldn't it be possible to analyze the frame type of the camera and if it contains RBG content then do all the call to be able to display the RGB image?
> 
> We could but it will add a few more conditions to be verified if the 3D camera is disabled.
That is not an impediment or is it?

**Comment by SeptimiuVana** [comment]
> > > There is a lot of code that is enabled only if smart 3D camera is enabled. Wouldn't it be possible to analyze the frame type of the camera and if it contains RBG content then do all the call to be able to display the RGB image?
> > 
> > 
> > We could but it will add a few more conditions to be verified if the 3D camera is disabled.
> 
> That is not an impediment or is it?

It will do extra checks for the other systems in which we do not have the rgb camera and it might affect the speed.

**Comment by dNechita** [comment]
> > > > There is a lot of code that is enabled only if smart 3D camera is enabled. Wouldn't it be possible to analyze the frame type of the camera and if it contains RBG content then do all the call to be able to display the RGB image?
> > > 
> > > 
> > > We could but it will add a few more conditions to be verified if the 3D camera is disabled.
> > 
> > 
> > That is not an impediment or is it?
> 
> It will do extra checks for the other systems in which we do not have the rgb camera and it might affect the speed.

Aditof-demo is an example that uses the SDK. The SDK exposes a generic API for a camera (it abstract over different types of cameras). This abstraction is one of the core features. If you use preprocessor directives to tell the demo which camera is using then what's the point in having a generic API? If we can't do it without directives then the API needs improving.
The demo should be able to ask the camera what are its capabilities and construct an UI according to that.
I don't think a few IFs should prevent us from doing this.

**Comment by SeptimiuVana** [comment]
> > > > > There is a lot of code that is enabled only if smart 3D camera is enabled. Wouldn't it be possible to analyze the frame type of the camera and if it contains RBG content then do all the call to be able to display the RGB image?
> > > > 
> > > > 
> > > > We could but it will add a few more conditions to be verified if the 3D camera is disabled.
> > > 
> > > 
> > > That is not an impediment or is it?
> > 
> > 
> > It will do extra checks for the other systems in which we do not have the rgb camera and it might affect the speed.
> 
> Aditof-demo is an example that uses the SDK. The SDK exposes a generic API for a camera (it abstract over different types of cameras). This abstraction is one of the core features. If you use preprocessor directives to tell the demo which camera is using then what's the point in having a generic API? If we can't do it without directives then the API needs improving.
> The demo should be able to ask the camera what are its capabilities and construct an UI according to that.
> I don't think a few IFs should prevent us from doing this.

Very good point, I will modify the code.

File: examples/aditof-demo/aditofdemoview.h (hunk 1 above)
Comment (SeptimiuVana) [reply examples/aditof-demo/aditofdemoview.h:102]: fixed

File: sdk/src/frame_impl.cpp [hunk 2]
```diff
@@ -130,8 +130,9 @@ aditof::Status FrameImpl::getData(aditof::FrameDataType dataType,
 }
 
 void FrameImpl::allocFrameData(const aditof::FrameDetails &details) {
-    m_fullData = new uint16_t[details.width * details.height * 2 +
-                              details.rgbWidth * details.rgbHeight];
+    //removed the usage of rgbWidth/height to avoid app crash when running on target
```
Comment (dNechita) [inline sdk/src/frame_impl.cpp:133]: What is this about?

**Review by dNechita — APPROVED** [review]
Rest looks good

File: sdk/src/frame_impl.cpp (hunk 2 above)
Comment (SeptimiuVana) [reply sdk/src/frame_impl.cpp]: Fixed. Found another way to disable the allocation

**Pull request merged**